src/helpers/figures/charts/gauge_chart_rendering.ts

```diff
@@ -0,0 +1,466 @@
+ */
+function getInflectionValues(
+  runtime: GaugeChartRuntime,
+  gaugeRect: Rect,
+  textColor: Color,
+  ctx: CanvasRenderingContext2D
+): InflectionValue[] {
+  const maxValue = runtime.maxValue;
+  const minValue = runtime.minValue;
+
+  const gaugeCircleCenter = {
+    x: gaugeRect.x + gaugeRect.width / 2,
+    y: gaugeRect.y + gaugeRect.height,
+  };
+  const textStyle = { fontSize: GAUGE_LABELS_FONT_SIZE };
+
+  const inflectionValues: InflectionValue[] = [];
+  const inflectionValuesTextRects: UnalignedRectangle[] = [];
+  for (const inflectionValue of runtime.inflectionValues) {
+    const percentage = (inflectionValue.value - minValue.value) / (maxValue.value - minValue.value);
+
+    const labelWidth = computeTextWidth(ctx, inflectionValue.label, textStyle, "px");
+    const angle = Math.PI - Math.PI * percentage;
+
+    const textRect = getRectangleTangentToCircle(
+      angle, // angle between X axis and the point where the rectangle is tangent to the circle
+      gaugeRect.height + GAUGE_INFLECTION_LABEL_BOTTOM_MARGIN, // radius of the gauge circle + margin below text
+      gaugeCircleCenter.x, // center of the gauge circle
+      gaugeCircleCenter.y, // center of the gauge circle
+      labelWidth + 2, // width of the text + some margin
+      GAUGE_LABELS_FONT_SIZE // height of the text
+    );
+    let offset = inflectionValuesTextRects.some((rect) => doRectanglesIntersect(rect, textRect))
+      ? GAUGE_LABELS_FONT_SIZE
+      : 0;
+    inflectionValuesTextRects.push(textRect);
+
+    inflectionValues.push({
+      rotation: angle,
+      label: inflectionValue.label,
+      fontSize: GAUGE_LABELS_FONT_SIZE,
+      color: textColor,
+      offset,
+    });
+  }
+
+  return inflectionValues;
+}
+
+function getGaugeColor(runtime: GaugeChartRuntime): Color {
+  const gaugeValue = runtime.gaugeValue?.value;
+  if (gaugeValue === undefined) {
+    return GAUGE_BACKGROUND_COLOR;
+  }
+  let colorIndex = 0;
+  while (runtime.inflectionValues[colorIndex]?.value <= gaugeValue) {
+    colorIndex++;
+  }
+  return runtime.colors[colorIndex];
+}
+
+function degreesToRadians(degrees: number) {
+  return (degrees * Math.PI) / 180;
+}
+
+function getContrastedTextColor(backgroundColor: Color) {
+  return relativeLuminance(backgroundColor) > 0.3
+    ? GAUGE_TEXT_COLOR
+    : GAUGE_TEXT_COLOR_HIGH_CONTRAST;
+}
+
+function getSegmentsOfRectangle(rectangle: UnalignedRectangle): Segment[] {
+  return [
+    { start: rectangle.topLeft, end: rectangle.topRight },
+    { start: rectangle.topRight, end: rectangle.bottomRight },
+    { start: rectangle.bottomRight, end: rectangle.bottomLeft },
+    { start: rectangle.bottomLeft, end: rectangle.topLeft },
+  ];
+}
+
+function doSegmentIntersect(segment1: Segment, segment2: Segment): boolean {
+  const A = segment1.start;
+  const B = segment1.end;
+  const C = segment2.start;
+  const D = segment2.end;
+
+  const angleSegment1 = Math.atan2(B.y - A.y, B.x - A.x);
```

Comment:
Not sure if we really need to use atan2 here. Maybe we can have something like 
`(B.y === A.y && C.y === D.y) || (B.x === A.x && C.x === D.x) || (B.y-A.y)/(B.x-A.x) === (D.y-C.y)/(C.x-D.x))` to check if they are colinear ?